### PR TITLE
Fix cache miss in fast-memoize when f returns nil

### LINKED ISSUE #123: Slowness when no content-type is provided
content-type is not required for GET requests. Not providing content type causes `fast-memoize` to always cache miss and run.
It is simple to fix, will provide a MR

### PR DESCRIPTION
**Pull request by bsless**
closes #123

### CODE REVIEW

File: modules/muuntaja/src/muuntaja/parse.clj [hunk 1]
```diff
@@ -7,21 +7,24 @@
...
+        unmiss (fn [x] (if (identical? x miss) nil x))
         cache! (fn [args]
-                 (let [value (apply f args)]
-                   (when value
-                     ;; not synchronized but ok enough
-                     (when (> (.size cache) size)
-                       (.clear cache))
-                     (.putIfAbsent cache args value))
+                 (let [value (apply f args)
+                       value (if (nil? value) miss value)]
```
Comment (ikitommi) [inline modules/muuntaja/src/muuntaja/parse.clj:17]: `value` could be just `(or (apply f args) miss)`

File: modules/muuntaja/src/muuntaja/parse.clj [hunk 2]
```diff
@@ -7,21 +7,24 @@
...
+                   (when (> (.size cache) size)
+                     (.clear cache))
+                   (.putIfAbsent cache args value)
                    value))]
     (fn [& args]
       (let [cached (.getOrDefault cache args sentinel)]
-        (if (identical? sentinel cached)
-          (cache! args)
-          cached)))))
+        (unmiss
```
Comment (ikitommi) [inline modules/muuntaja/src/muuntaja/parse.clj:24]: could be unfolded as: 

```clj
        (cond
          (identical? cached sentinel) (cache! args)
          (identical? cached miss) nil
          :else cached)
```

**Review by ikitommi — APPROVED** [review]
looks good.

**Comment by bsless** [comment]
@ikitommi Done both suggested comments. I see there's no way to avoid checking for identity with `miss` in `cache!` as well.
Let me if you think this should be reorganized

**Comment by ikitommi** [comment]
Thanks!

**Pull request merged**